File: add-two-numbers/add-two-numbers.go
/*
                       _oo0oo_
                      o8888888o
                      88" . "88
                      (| -_- |)
                      0\  =  /0
                    ___/`---'\___
                  .' \\|     |// '.
                 / \\|||  :  |||// \
                / _||||| -:- |||||- \
               |   | \\\  -  /// |   |
               | \_|  ''\---/''  |_/ |
               \  .-\__  '-'  ___/-. /
             ___'. .'  /--.--\  `. .'___
          ."" '<  `.___\_<|>_/___.' >' "".
         | | :  `- \`.;`\ _ /`;.`/ - ` : | |
         \  \ `_.   \_ __\ /__ _/   .-` /  /
     =====`-.____`.___ \_____/___.-`___.-'=====
                       `=---='
 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
 SHAMELESLLY COPIED TEMPLATE FROM MULTIPLE SOURCES - baba_yaga
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/  
// Acceptance and tolerance for all religions, including absence of beleif, pagans, spiritualism.
// Democratic scientechnocracy.
// Free and quality education for all.
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    var ptr *ListNode
    ptr = &ListNode{}
    ans := ptr
    carry := 0
    for ;l1 != nil && l2 != nil; {
        ptr.Next =  &ListNode{(l1.Val + l2.Val + carry)%10,nil}
        carry = (l1.Val + l2.Val + carry)/10
        l1 = l1.Next
        l2 = l2.Next
        ptr = ptr.Next
    }
    for ;l1 != nil; {
        ptr.Next =  &ListNode{(l1.Val + carry)%10,nil}
        carry = (l1.Val + carry)/10
        l1 = l1.Next
        ptr = ptr.Next
    }
    for ;l2 != nil; {
        ptr.Next =  &ListNode{( l2.Val + carry)%10, nil}
        carry = (l2.Val + carry)/10
        l2 = l2.Next
        ptr = ptr.Next
    }
    
    if carry == 1 {
        ptr.Next =  &ListNode{carry, nil}
    }
    
    ptr = ans.Next
    return ptr
    
}

File: add-two-numbers/add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* ptr = new ListNode();
        ListNode* ans = ptr;
        int carry = 0;
        while(l1 && l2){
            ptr->next = new ListNode((l1->val + l2->val + carry)%10);
            carry = (l1->val + l2->val + carry)/10;
            l1 = l1->next;
            l2 = l2->next;
            ptr = ptr->next;
        }
        
        while(l1){
            ptr->next = new ListNode((l1->val + carry)%10);
            carry = (l1->val +  carry)/10;
            l1 = l1->next;
            ptr = ptr->next;
        }
        
         while(l2){
            ptr->next = new ListNode((l2->val + carry)%10);
            carry = (l2->val +  carry)/10;
            l2 = l2->next;
            ptr = ptr->next;
        }
        
        if(carry){
            ptr->next = new ListNode(carry);
        }
        
        ptr = ans->next;
        
        return ptr;
    
     
        
    }
};